### PR TITLE
Disable tracing on empty `traces_sample_rate`

### PR DESCRIPTION
**Pull request by stayallive**
Noticed we can probably disable the tracing completely if the `traces_sample_rate` is `0.0` since there would be no chance to send a trace to Sentry which would negate any performance impact the tracing code could have on an application.

Looking at this code do we want to collect the trace data anyway even if we already know the sample rate is 0 and thus the trace will not be sent or is the `traces_sampler` option supposed to be able to override that later? Because then this PR might not be good...

Bottom line, if the `traces_sample_rate === 0.0` should we even collect tracing data?
Yes = Merge, No = Close 😉

### CODE REVIEW

**Comment by HazAT** [comment]
Tricky, PHP is a bit unique in that regard since it basically has no persistence in any way between requests. So it should be ok to do it. But let's wait a bit until we gathered more feedback to release this.

**Comment by stayallive** [comment]
After consideration and looking at real world performance data I have decided to postpone this until actually needed since currently the impact is close to 0 and the impact with all the ways tracing can be sampled it's not worth it right now.
